rs/replicated_state/src/canister_state/system_state.rs

```diff
@@ -428,77 +467,57 @@ impl TaskQueue {
-        });
+        self.queue.clear();
     }
 
-    /// Returns all `PausedExecution` and `PausedInstallCode` tasks.
-    pub fn get_all_paused_tasks(&self) -> Vec<&ExecutionTask> {
-        let mut res = Vec::new();
-        for task in self.queue.iter() {
+    /// Returns `PausedExecution` or `PausedInstallCode` task.
+    pub fn get_paused_task(&self) -> Option<&ExecutionTask> {
+        if let Some(task) = &self.paused_or_aborted_task {
             match task {
-                ExecutionTask::PausedExecution { .. } | ExecutionTask::PausedInstallCode(..) => {
-                    res.push(task)
+                ExecutionTask::PausedExecution { .. } | ExecutionTask::PausedInstallCode(_) => {
+                    Some(task)
                 }
+                ExecutionTask::AbortedExecution { .. }
+                | ExecutionTask::AbortedInstallCode { .. } => None,
                 ExecutionTask::Heartbeat
                 | ExecutionTask::GlobalTimer
-                | ExecutionTask::AbortedExecution { .. }
-                | ExecutionTask::AbortedInstallCode { .. }
-                | ExecutionTask::OnLowWasmMemory => (),
+                | ExecutionTask::OnLowWasmMemory => unreachable!(
+                    "Unexpected on task type in the in TaskQueue::paused_or_aborted_task."
+                ),
             }
+        } else {
+            None
         }
-        res
     }
 
-    /// Replace all `PausedExecution` and `PausedInstallCode` with corresponding
-    /// `AbortedExecution` and `AbortedInstallCode` respectively.
-    pub fn replace_paused_with_aborted_tasks(
-        &mut self,
-        mut aborted_tasks: HashMap<PausedExecutionId, ExecutionTask>,
-    ) {
-        let task_queue = std::mem::take(&mut self.queue);
-
-        self.queue = task_queue
-            .into_iter()
-            .map(|task| match task {
-                ExecutionTask::AbortedExecution { .. }
-                | ExecutionTask::AbortedInstallCode { .. }
-                | ExecutionTask::Heartbeat
-                | ExecutionTask::GlobalTimer
-                | ExecutionTask::OnLowWasmMemory => task,
-                ExecutionTask::PausedExecution { id, .. } => {
-                    let aborted = aborted_tasks.remove(&id).unwrap();
-                    debug_assert!(matches!(aborted, ExecutionTask::AbortedExecution { .. }));
-                    aborted
-                }
-                ExecutionTask::PausedInstallCode(id) => {
-                    let aborted = aborted_tasks.remove(&id).unwrap();
-                    debug_assert!(matches!(aborted, ExecutionTask::AbortedInstallCode { .. }));
-                    aborted
-                }
-            })
-            .collect();
+    /// Replace `PausedExecution` or `PausedInstallCode` with corresponding
+    /// `AbortedExecution` or `AbortedInstallCode` respectively.
+    pub fn replace_paused_with_aborted_task(&mut self, aborted_task: ExecutionTask) {
+        debug_assert!(match &aborted_task {
```

Comment:
I did, but since we will have `debug_assert` and `unreachable` in the same match statement, I changed `debug_assert` to `assert` and added additional tests.